api/app/db/crud.py

```diff
@@ -219,7 +220,7 @@ def get_historic_station_model_predictions(
                PredictionModel.abbreviation == model).\
         order_by(WeatherStationModelPrediction.station_code).\
         order_by(WeatherStationModelPrediction.prediction_timestamp).\
-        order_by(PredictionModelRunTimestamp.prediction_run_timestamp.desc())
+        order_by(PredictionModelRunTimestamp.prediction_run_timestamp.asc())
```

Comment:
Doing it this way, when looping through records, if there's more than one prediction for any point in time, then older predictions are just naturally overwritten by newer predictions.

What really should be happening is some fancy nested query so that the result contains only what is needed.

Reply:
Ahhh ok I think I understand your reasoning for this now. I ask because I believe I had it returning in descending order for the purpose of fetching the most recent historic model, so the first returned prediction for a given timestamp would be the most recently issued prediction. (But I did also insert redundant code to check that only the most recent prediction is used, so I don't think this change will cause any problems with the most recent historic stuff.)

Agree that it would be ideal to find a nice way to query the DB to only get the results that we want.